akka-contrib/src/main/scala/akka/contrib/pattern/TransparentExponentialBackoffSupervisor.scala

```diff
@@ -0,0 +1,192 @@
+   * @param minBackoff the min time before the child is restarted.
+   * @param maxBackoff the max time (upperbound) for a child restart.
+   * @param randomFactor a random delay factor to add on top of the calculated exponential
+   *   back off.
+   *   The calculation is equivalent to:
+   *   {{{
+   *     final_delay = min(
+   *       maxBackoff,
+   *       (random_delay_factor * calculated_backoff) + calculated_backoff)
+   *   }}}
+   * @param decider an [[akka.actor.SupervisorStrategy.Decider]] to specify how the supervisor
+   *   should behave for different exceptions. If no cases are matched, the default decider of
+   *   [[akka.actor.Actor]] is used. When the [[akka.actor.SupervisorStrategy.Restart]] directive
+   *   is returned by the decider, this supervisor will apply an exponential back off restart.
+   */
+  def propsWithDecider(
+    childProps: Props,
+    minBackoff: FiniteDuration,
+    maxBackoff: FiniteDuration,
+    randomFactor: Double)(decider: Decider): Props = {
+    Props(
+      classOf[TransparentExponentialBackoffSupervisor],
+      childProps,
+      Some(decider),
+      minBackoff,
+      maxBackoff,
+      randomFactor)
+  }
+
+  /**
+   * Props for creating a [[TransparentExponentialBackoffSupervisor]] using the
+   * default [[akka.actor.Actor]] decider.
+   *
+   * @param childProps the [[akka.actor.Props]] of the child to be supervised.
+   * @param minBackoff the min time before the child is restarted.
+   * @param maxBackoff the max time (upperbound) for a child restart.
+   * @param randomFactor a random delay factor to add on top of the calculated exponential
+   *   back off.
+   *   The calculation is equivalent to:
+   *   {{{
+   *     final_delay = min(
+   *       maxBackoff,
+   *       (random_delay_factor * calculated_backoff) + calculated_backoff)
+   *   }}}
+   */
+  def props(
+    childProps: Props,
+    minBackoff: FiniteDuration,
+    maxBackoff: FiniteDuration,
+    randomFactor: Double): Props = {
+    Props(
+      classOf[TransparentExponentialBackoffSupervisor],
+      childProps,
+      None,
+      minBackoff,
+      maxBackoff,
+      randomFactor)
+  }
+}
+
+/**
+ * A supervising actor that restarts a child actor with an exponential back off.
+ *
+ * This explicit supervisor behaves similarly to the normal implicit supervision where
+ * if an actor throws an exception, the decider on the supervisor will decide when to
+ * `Stop`, `Restart`, `Escalate`, `Resume` the child actor.
+ *
+ * When the `Restart` directive is specified, the supervisor will delay the restart
+ * using an exponential back off strategy (bounded by minBackoff and maxBackoff).
+ *
+ * This supervisor is intended to be transparent to both the child actor and external actors.
+ * Where external actors can send messages to the supervisor as if it was the child and the
+ * messages will be forwarded. And when the child is `Terminated`, the supervisor is also
+ * `Terminated`.
```

Comment:
why can't we use the same semantics as the old `BackoffSupervisor` so that this can be used with Akka Persistence, i.e. when the child actor is terminated exponential back-off kicks in?